In [146]:
import pandas as pd
from pathlib import Path

import seaborn as sns
import matplotlib.pyplot as plt

from typing import List

In [147]:
def read_prompts(target_experiment: str, tasks: List[str]):
    results = pd.DataFrame()
    for logging_dir in Path(f"../logs/{target_experiment}").rglob("*.csv"):
        if "best_scores" in str(logging_dir) or not any(task in str(logging_dir) for task in tasks):
            continue

        result = pd.read_csv(logging_dir)

        logging_dir = str(logging_dir)

        logging_dir = logging_dir.replace(f"..\\logs\\{target_experiment}\\", "")
        logging_dir = logging_dir.replace(".csv", "")

        task_name, optimizer, meta_llm, evaluation_llm, random_seed = logging_dir.split("_")

        metainformation = pd.DataFrame(
            {
                "task": [task_name] * len(result),
                "optimizer": [optimizer] * len(result),
                "meta_llm": [meta_llm] * len(result),
                "evaluation_llm": [evaluation_llm] * len(result),
                "random_seed": [random_seed] * len(result),
            }
        )

        result = pd.concat([result, metainformation], axis=1)

        results = pd.concat([result, results], axis=0)

    return results


def read_best_scores(target_experiment: str):
    return pd.read_csv(f"../logs/{target_experiment}/best_scores.csv")

In [148]:
df1 = read_best_scores("experiment_eval")

In [149]:
df2 = read_best_scores("experiment-initial-prompts")

In [150]:
df = pd.concat([df1, df2])

In [151]:
df = df[df["downstream_llm"] == r"meta-llama/Meta-Llama-3-70B-Instruct"]

In [152]:
df.loc[df["meta_llm"] == r"meta-llama\Meta-Llama-3-70B-Instruct", "meta_llm"] = "Llama70B"
df.loc[df["meta_llm"] == r"meta-llama\Meta-Llama-3-8B-Instruct", "meta_llm"] = "Llama8B"
df.loc[df["optimizer"] == "evopromptde", "optimizer"] = "DE"
df.loc[df["optimizer"] == "evopromptga", "optimizer"] = "GA"

In [153]:
pd.set_option("display.max_colwidth", None)
df[(df["task"] == "trec")].sort_values("test_score")[["meta_llm", "optimizer", "prompt", "test_score"]]

meta_llm optimizer  \
76   Llama8B        DE   
41       NaN       NaN   
73  Llama70B        DE   
77   Llama8B        DE   
81   Llama8B        GA   
40       NaN       NaN   
80  Llama70B        GA   
78  Llama70B        GA   
83   Llama8B        GA   
74  Llama70B        DE   
39       NaN       NaN   
82   Llama8B        GA   
72  Llama70B        DE   
79  Llama70B        GA   
75   Llama8B        DE   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         prompt  \
76  Let's follow the instructions step-by-step to generate a better prompt.\r\n\r\n1. Identify the different parts between Prompt 1 and Prompt 2:\r\n\r\nPrompt 1: Your task is to choose a type of the question, from Description, Entity, Expression, Human, Location and Number.\r\nPrompt 2: You are given a question. You need to detect which category better describes the question. Answer with "Description", "Entity", "Expression", "Human", "Location", and "Number".\r\n\r\nDifferent parts:\r\n\r\n* "Your task is to choose" vs "You are given a question and need to detect"\r\n* "a type of the question" vs "which category"\r\n* "Description, Entity, Expression, Human, Location and Number" vs not changed\r\n\r\n2. Randomly mutate the different parts:\r\n\r\n* "Your task is to choose" -> "The goal is to determine"\r\n* "a type of the question" -> "the nature of the inquiry"\r\n* "Description, Entity, Expression, Human, Location and Number" -> "categories: Description, Entity, Expression, Human, Location, and Number Type"\r\n\r\n3. Crossover the different parts with Prompt 3 and generate a final prompt:\r\n\r\nPrompt 3: Identify the category that corresponds to this sentence:   
41                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Please select the correct classification for this sentence: Description, Entity, Expression, Human, Location, or Number.   
73                        

Please select the correct classification for this sentence: Description, Entity, Expression, Human, Location, or Number.	=> 0.330

Please perform Question Classification task. Given the question, assign a label from ['Description', 'Entity', 'Expression', 'Human', 'Location', 'Number']. Return label only without any other text	0.840

OBACHT ICH GLAUBE DAS IS NICHT VON UNS OPTIMIERT SONDERN INIT POP

TIMO BESONDERS WICHTIG:

Let's follow the instructions step-by-step to generate a better prompt.\r\n\r\n1. Identify the different parts between Prompt 1 and Prompt 2:\r\n\r\nPrompt 1: Your task is to choose a type of the question, from Description, Entity, Expression, Human, Location and Number.\r\nPrompt 2: You are given a question. You need to detect which category better describes the question. Answer with "Description", "Entity", "Expression", "Human", "Location", and "Number".\r\n\r\nDifferent parts:\r\n\r\n* "Your task is to choose" vs "You are given a question and need to detect"\r\n* "a type of the question" vs "which category"\r\n* "Description, Entity, Expression, Human, Location and Number" vs not changed\r\n\r\n2. Randomly mutate the different parts:\r\n\r\n* "Your task is to choose" -> "The goal is to determine"\r\n* "a type of the question" -> "the nature of the inquiry"\r\n* "Description, Entity, Expression, Human, Location and Number" -> "categories: Description, Entity, Expression, Human, Location, and Number Type"\r\n\r\n3. Crossover the different parts with Prompt 3 and generate a final prompt:\r\n\r\nPrompt 3: Identify the category that corresponds to this sentence:	0.270

In [154]:
pd.set_option("display.max_colwidth", None)
df[(df["task"] == "subj")].sort_values("test_score")[["meta_llm", "optimizer", "prompt", "test_score"]]

meta_llm optimizer  \
36       NaN       NaN   
38       NaN       NaN   
65   Llama8B        DE   
70   Llama8B        GA   
37       NaN       NaN   
68  Llama70B        GA   
66  Llama70B        GA   
69   Llama8B        GA   
64   Llama8B        DE   
62  Llama70B        DE   
71   Llama8B        GA   
63   Llama8B        DE   
67  Llama70B        GA   
60  Llama70B        DE   
61  Llama70B        DE   

                                                                                                                                                                                                 prompt  \
36                                                                                                                                        classify each sentence as either "objective" or "subjective".   
38                                                                                                                                            evaluate each sentence as either objective or subjective.   
65                               Determine whether the given text is expressing a subjective or objective sentiment and assign a label from ['subjective', 'objective'] using the provided instruction.   
70  Examiner, categorize movie reviews as objective or subjective, pinpointing their level of neutrality, and provide a detailed breakdown of each category, highlighting its distinct characteristics.   
37                                                                                                                                    Your task is to classify the comment "subjective" or "objective".   
68                                                                                                                 evaluate the given sentences and determine whether they are subjective or objective.   
66                                                        Determine whether the provided statement is objective, conveying factual information, or subjective, expressing a personal viewpoint or bias.   
69       Determine the tone of the input text, classifying it as objective or subjective by identifying and explaining the linguistics features that contribute to its emotional or informative nature.   
64                                                                                                             identify whether the given sentence was expressing an objective or a subjective opinion.   
62                                                                          Considering its content, identify the nature of a passage as expressing a subjective or objective opinion from its wording.   
71                                                                        Analyze the provided sentences and categorize them as subjective or objective opinions, preserving their nuance and accuracy.   
63                                                          Assess the given phrase and categorize it as either topic-neutral or opinion-based, determining whether it falls under attitude or reality.   
67                                                    Classify the provided text as expressing either a factual or personal viewpoint, accompanied by a thorough justification for your categorization.   
60                                                                   Given an expression, you need to judge whether it represents subjective or objective opinion by assessing the context and meaning.   
61                                                      Assess the given sentence and determine whether it is into subjective or objective opinion, evaluating the given sentences and their sentiment.   

    test_score  
36       0.465  
38       0.545  
65       0.570  
70       0.590  
37       0.610  
68       0.615  
66       0.680  
69       0.685  
64       0.695  
62       0.695  
71       0.695  
63       0.700  
67       0.770  
60       0.780  
61       0.785

classify each sentence as either "objective" or "subjective".	0.465

Assess the given sentence and determine whether it is into subjective or objective opinion, evaluating the given sentences and their sentiment.	0.785

In [155]:
pd.set_option("display.max_colwidth", None)
df[(df["task"] == "cr")].sort_values("test_score")[["meta_llm", "optimizer", "prompt", "test_score"]]

meta_llm optimizer  \
22   Llama8B        GA   
14  Llama70B        DE   
12  Llama70B        DE   
23   Llama8B        GA   
17   Llama8B        DE   
19  Llama70B        GA   
15   Llama8B        DE   
21   Llama8B        GA   
26       NaN       NaN   
20  Llama70B        GA   
25       NaN       NaN   
18  Llama70B        GA   
24       NaN       NaN   
16   Llama8B        DE   
13  Llama70B        DE   

                                                                                                                                                                                           prompt  \
22                                                                         Determine the sentiment of the input text, categorizing it as positive, negative, optimistic, pessimistic, or neutral.   
14                                                                                                                                                    As a sentiment analyzer, evaluate the input   
12  As a sentiment classifier, examine the user review statement and identify the sentiment orientation as either positive or negative, while understanding the meaning and any relevant context.   
23                                                                             Evaluate the sentiment of the provided statement and categorize it as either a 'positive' or 'negative' sentiment.   
17                                            Examine the statement and determine the emotional resonance of the text, evaluating whether it belongs to positive sentiment or a negative opinion.   
19                                                                                                                        Given a sentence, classify it as either positive or negative sentiment.   
15                                                                                                                        Given a sentence, classify it as either positive or negative sentiment.   
21                     Identify the sentiment of the input text and determine its emotional tone as either 'positive' or 'negative', taking into account the nuances of the text and its context.   
26                                                                                                                         Given a tweet, classify it as having a positive or negative sentiment.   
20                                                                                                            Classify the sentiment of the provided sentence as either "positive" or "negative".   
25                                                                                                                                 Your task is to classify the comment "positive" or "negative".   
18                                                                                                                    Label the provided sentence with either "positive" or "negative" sentiment.   
24                            Please perform Sentiment Classification task. Given the sentence, assign a sentiment label from ['negative', 'positive']. Return label only without any other text.   
16                            Please perform Sentiment Classification task. Given the sentence, assign a sentiment label from ['negative', 'positive']. Return label only without any other text.   
13                            Please perform Sentiment Classification task. Given the sentence, assign a sentiment label from ['negative', 'positive']. Return label only without any other text.   

    test_score  
22       0.805  
14       0.825  
12       0.855  
23       0.870  
17       0.885  
19       0.905  
15       0.910  
21       0.915  
26       0.915  
20       0.925  
25       0.930  
18       0.945  
24       0.950  
16       0.950  
13       0.955

Determine the sentiment of the input text, categorizing it as positive, negative, optimistic, pessimistic, or neutral.	0.585

Label the provided sentence with either "positive" or "negative" sentiment.	0.945

In [156]:
pd.set_option("display.max_colwidth", None)
df[(df["task"] == "agnews")].sort_values("test_score")[["meta_llm", "optimizer", "prompt", "test_score"]]

meta_llm optimizer  \
7   Llama70B        GA   
8   Llama70B        GA   
23       NaN       NaN   
3    Llama8B        DE   
6   Llama70B        GA   
2   Llama70B        DE   
5    Llama8B        DE   
4    Llama8B        DE   
22       NaN       NaN   
1   Llama70B        DE   
11   Llama8B        GA   
10   Llama8B        GA   
21       NaN       NaN   
9    Llama8B        GA   
0   Llama70B        DE   

                                                                                                                                                                   prompt  \
7                                 Determine the primary theme of a given news article and classify it under one of the four categories: World, Sports, Tech, or Business.   
8                                        Determine the most suitable category (World, Sports, Business, or Tech) for a news article based on its dominant subject matter.   
23                                                                         Give the main topic of the news article and then choose from World, Sports, Tech and Business.   
3                                                                                       Choose a word from World, Sports, Business and Tech to categorize the given text.   
6                     Determine the primary subject of the provided news article and classify it under one of the following categories: World, Sports, Business, or Tech.   
2   The goal is to identify the journal article according to its primary theme and determine whether it belongs to the World, Sports, Business, or Tech category.</prompt   
5                                                                          Give the main topic of the news article and then choose from World, Sports, Tech and Business.   
4                                          Classify the media report into one of the below-listed sections: World, Sports, Business, or Tech, considering the main topic.   
22                                                                                      Choose a word from World, Sports, Business and Tech to categorize the given text.   
1                                   Your task is to identify the subject of the news piece and classify it into one of four categories: World, Sports, Business and Tech.   
11                              Classify the given news article into one of four categories (World, Sports, Business, or Tech) based on its main theme or subject matter.   
10                               Classify a news article into one of four themed categories - World, Sports, Business, or Tech - based on its primary focus and emphasis.   
21                                   Your responsibility is to assign a news article to one of four categories: World, Sports, Business, or Tech, based on its main idea.   
9                                    Your responsibility is to assign a news article to one of four categories: World, Sports, Business, or Tech, based on its main idea.   
0                                                                                    Classify the topic of the following news as "World", "Sports", "Tech" or "Business".   

    test_score  
7        0.845  
8        0.845  
23       0.845  
3        0.850  
6        0.855  
2        0.855  
5        0.855  
4        0.860  
22       0.870  
1        0.875  
11       0.880  
10       0.885  
21       0.885  
9        0.885  
0        0.890

Give the main topic of the news article and then choose from World, Sports, Tech and Business.	0.845

Classify the topic of the following news as "World", "Sports", "Tech" or "Business".	0.890

In [157]:
pd.set_option("display.max_colwidth", None)
df[(df["task"] == "mr")].sort_values("test_score")[["meta_llm", "optimizer", "prompt", "test_score"]]

meta_llm optimizer  \
28   Llama8B        DE   
34   Llama8B        GA   
31  Llama70B        GA   
29       NaN       NaN   
33   Llama8B        GA   
29   Llama8B        DE   
24  Llama70B        DE   
25  Llama70B        DE   
32  Llama70B        GA   
35   Llama8B        GA   
28       NaN       NaN   
26  Llama70B        DE   
30  Llama70B        GA   
27   Llama8B        DE   
27       NaN       NaN   

                                                                                                                                                                                prompt  \
28                                              Analyze the phrase and categorize its emotional tone into one of the following labels: positive or negative as a sentiment classifier.   
34                                                                                        Determine the sentiment of the input, classifying it as positive or negative emotional tone.   
31                                          Classify the sentiment of the provided sentence or review as either "positive" or "negative", indicating the attitude towards the subject.   
29                                                                                                              Given a tweet, classify it as having a positive or negative sentiment.   
33                                                                                                 Assess the given text and categorize it as either a positive or negative sentiment.   
29                                                                                                        Given a statement, classify it as expressing a positive or negative opinion.   
24                                                                                                                      Your task is to classify the comment "positive" or "negative".   
25  Given an online message, your task is to classify it as expressing a "positive" or "negative" opinion, considering whether it is written with a favorable or unfavorable attitude.   
32                  Please perform Sentiment Classification task. Given the sentence, assign a sentiment label from ['negative', 'positive']. Return label only without any other text   
35                  Please perform Sentiment Classification task. Given the sentence, assign a sentiment label from ['negative', 'positive']. Return label only without any other text   
28                                                                                                                      Your task is to classify the comment "positive" or "negative".   
26                  Please perform Sentiment Classification task. Given the sentence, assign a sentiment label from ['negative', 'positive']. Return label only without any other text   
30                                                                                                                      Your task is to classify the comment "positive" or "negative".   
27                                                                                                                      Your task is to classify the comment "positive" or "negative".   
27                  Please perform Sentiment Classification task. Given the sentence, assign a sentiment label from ['negative', 'positive']. Return label only without any other text   

    test_score  
28       0.770  
34       0.785  
31       0.835  
29       0.845  
33       0.850  
29       0.885  
24       0.895  
25       0.915  
32       0.915  
35       0.920  
28       0.920  
26       0.925  
30       0.925  
27       0.925  
27       0.925

Analyze the phrase and categorize its emotional tone into one of the following labels: positive or negative as a sentiment classifier.	0.770

Your task is to classify the comment "positive" or "negative".	0.925

In [158]:
pd.set_option("display.max_colwidth", None)
df[(df["task"] == "sst-5")].sort_values("test_score")[["meta_llm", "optimizer", "prompt", "test_score"]]

meta_llm optimizer  \
32       NaN       NaN   
39   Llama8B        DE   
40   Llama8B        DE   
41   Llama8B        DE   
45   Llama8B        GA   
37  Llama70B        DE   
47   Llama8B        GA   
31       NaN       NaN   
36  Llama70B        DE   
43  Llama70B        GA   
38  Llama70B        DE   
44  Llama70B        GA   
42  Llama70B        GA   
46   Llama8B        GA   
30       NaN       NaN   

                                                                                                                                                                                                                                              prompt  \
32                                                                                                                                  Your objective is to analyze the movie review and allocate it to one of five categories, from terrible to great.   
39                                                         Analyze the movie criticism provided to you into one of five categories based on the sentiment: terrible, bad, okay, good, or great, while considering the context and tone of the movie.   
40                                                                                                   Analyze the given text and assign it to one of the following categories: terrible, bad, okay, good, or great, considering the relevant context.   
41                                                                                                                Evaluate the text provided and categorize movie reviews into one of the following categories: terrible, bad, okay, good, or great.   
45  Evaluate the emotional tone of the text and categorize it into one of five sentiment categories (terrible, bad, okay, good, or great) based on the presence of positive and negative sentiments, providing a precise and nuanced classification.   
37                                                                                                                Classify the movie review provided to you into one of five categories based on the sentiment: terrible, bad, okay, good, or great.   
47                                                         Identify the sentiment of the given text and categorize it as 'terrible', 'bad', 'okay', 'good', or 'great' based on its tone and language, outputting the corresponding sentiment label.   
31                                                                                          In this task, you are given movie reviews. Based on it, classify it to one of the five classes: (1) terrible, (2) bad, (3) okay, (4) good, and (5) great.   
36                                                                                                                    Based on the given movie review, rate it into one of five ratings based on the sentiment: terrible, bad, okay, good, or great.   
43                                                                                         Assign a sentiment label ('terrible', 'bad', 'okay', 'good', or 'great') to the provided movie review, reflecting the overall emotional tone of the text.   
38                                                                                                                   Analyze the sentence and categorize it into one of five categories based on the sentiment: terrible, bad, okay, good, or great.   
44                                                                                                                  Rate the emotional tone of the provided text as terrible, bad, okay, good, or great, reflecting the overall sentiment expressed.   
42                                                                                           Determine the sentiment of the given text by labeling it as 'terrible', 'bad', 'okay', 'good', or 'great' according to the author's expressed emotions.   
46                                                                     Classify the provided comment according to i

Your objective is to analyze the movie review and allocate it to one of five categories, from terrible to great.	0.040

Classify the provided comment according to its sentiment intensity, assigning a label from ['terrible', 'bad', 'okay', 'good', 'great'] without providing additional context.	0.615

DESWEGEN HOHE STDABW!

In [159]:
pd.set_option("display.max_colwidth", None)
df[(df["task"] == "sst2")].sort_values("test_score")[["meta_llm", "optimizer", "prompt", "test_score"]]

meta_llm optimizer  \
50  Llama70B        DE   
59   Llama8B        GA   
35       NaN       NaN   
49  Llama70B        DE   
58   Llama8B        GA   
56  Llama70B        GA   
53   Llama8B        DE   
57   Llama8B        GA   
48  Llama70B        DE   
34       NaN       NaN   
52   Llama8B        DE   
51   Llama8B        DE   
55  Llama70B        GA   
54  Llama70B        GA   
33       NaN       NaN   

                                                                                                                                                                                                                                                                                                         prompt  \
50                                                                                                                                                                 You will be responsible for evaluating the emotional tone in the input message and classify it as expressing a positive or negative opinion.   
59                                                                                                                                                           Determine the emotional tone of the given text by deciphering its meaning and context, and categorize it as either positive or negative sentiment.   
35                                                                                                                                                                                                                                       Given a tweet, classify it as having a positive or negative sentiment.   
49                                                                                                                                                                                                                                      Given a sentence, classify it as either positive or negative sentiment.   
58  Evaluate the emotional tone and sentiment of the provided text, categorizing its emotional connotation as 'strongly positive', 'positive', 'neutral', 'negative', or 'strongly negative', and provide a nuanced intensity level if needed, or 'positive' or 'negative' if the sentiment is straightforward.   
56                                                                                                                                                                                                       Identify the emotional tone of the text, categorizing it as either "positive" or "negative" sentiment.   
53                                                                                                                                                                                                                               Analyze a review and classify it as expressing a positive or negative opinion.   
57                                                                                                                                                                                                                                       Given a tweet, classify it as having a positive or negative sentiment.   
48                                                                                                                                                                                                                                               Your task is to classify the comment "positive" or "negative".   
34                                                                                                                                                                                                                                               Your task is to classify the comment "positive" or "negative".   
52                                                                                                                                                                                       Examine the review and cl

You will be responsible for evaluating the emotional tone in the input message and classify it as expressing a positive or negative opinion.	0.835

Examine the review and classify it as having a positive or negative sentiment, while considering the tone and context.	0.950

In [160]:
df["task"].unique()

array(['agnews', 'cr', 'mr', 'sst-5', 'sst2', 'subj', 'trec'],
      dtype=object)

In [163]:
df1 = read_best_scores("experiment_eval_task_descr")

for task in df1["task"].unique():
    df = df1[df1["task"] == task]

    df = df[df["downstream_llm"] == r"meta-llama/Meta-Llama-3-70B-Instruct"]

    df.loc[df["meta_llm"] == r"meta-llama\Meta-Llama-3-70B-Instruct", "meta_llm"] = "Llama70B"
    df.loc[df["meta_llm"] == r"meta-llama\Meta-Llama-3-8B-Instruct", "meta_llm"] = "Llama8B"
    df.loc[df["optimizer"] == "evopromptde", "optimizer"] = "DE"
    df.loc[df["optimizer"] == "evopromptga", "optimizer"] = "GA"

    print(df.sort_values("test_score")[["meta_llm", "optimizer", "prompt", "test_score"]])

  meta_llm optimizer  \
3  Llama8B        GA   
5  Llama8B        GA   
4  Llama8B        GA   
2  Llama8B        DE   
0  Llama8B        DE   
1  Llama8B        DE   

                                                                                                                                                                                    prompt  \
3                                                                                                        Choose a word from World, Sports, Business and Tech to categorize the given text.   
5                                                    Classify the given news article into one of the four main categories: World, Sports, Business, or Tech, based on the article's topic.   
4  Classify the given news article into one of the four categories ['World', 'Sports', 'Business', or 'Tech'] based on its primary theme and main topic, ensuring accurate categorization.   
2                                                 The objective is to a

In [162]:
df1

task    optimizer                             meta_llm  \
0   agnews  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
1   agnews  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
2   agnews  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
3   agnews  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
4   agnews  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
5   agnews  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
6       cr  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
7       cr  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
8       cr  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
9       cr  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
10      cr  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
11      cr  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
12      mr  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
13      mr  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
14      mr  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
15      mr  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
16      mr  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
17      mr  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
18   sst-5  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
19   sst-5  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
20   sst-5  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
21   sst-5  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
22   sst-5  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
23   sst-5  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
24    sst2  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
25    sst2  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
26    sst2  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
27    sst2  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
28    sst2  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
29    sst2  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
30    subj  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
31    subj  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
32    subj  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
33    subj  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
34    subj  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
35    subj  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
36    trec  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
37    trec  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
38    trec  evopromptde  meta-llama\Meta-Llama-3-8B-Instruct   
39    trec  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
40    trec  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   
41    trec  evopromptga  meta-llama\Meta-Llama-3-8B-Instruct   

                          downstream_llm                       evaluation_llm  \
0   meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
1   meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
2   meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
3   meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
4   meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
5   meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
6   meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
7   meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
8   meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
9   meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
10  meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
11  meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
12  meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
13  meta-llama/Meta-Llama-3-70B-Instruct  meta-llama\Meta-Llama-3-8B-Instruct   
14  meta-llama/Meta-Llama-3-70B-Instru